python/metatensor-operations/tests/block_from_array.py

```diff
@@ -60,6 +60,40 @@ def test_with_components():
     )
 
 
+@pytest.mark.parametrize("sample_names", [None, ["a"], ["a", "b"]])
+@pytest.mark.parametrize("property_names", [None, ["A"], ["A", "B"]])
+def test_with_label_names(sample_names, property_names):
+    """Test block_from_array with explicit sample and property names."""
+    array = array = np.zeros((3, 2, 1, 2, 3))
+    block = metatensor.block_from_array(
+        array, sample_names=sample_names, property_names=property_names
+    )
+
+    if sample_names is None:
+        sample_names = ["sample"]
+    if property_names is None:
+        property_names = ["property"]
+    expected_shape = (
+        np.prod(
+            array.shape[: len(sample_names)],
+        )
+        + tuple(
+            array.shape[i]
+            for i in range(len(sample_names), len(array.shape) - len(property_names))
+        )
+        + (len(property_names),)
+    )
+    assert (s1 == s2 for (s1, s2) in zip(block.values.shape, expected_shape))
```

Comment:
This works? I would have gone for 

```suggestion
    for actual_shape, expected in zip(block.values.shape, expected_shape)
        assert actual_shape == expected
```